base/client/src/main/java/dev/rico/internal/client/projector/JavaFXProjectorImpl.java

```diff
@@ -97,6 +93,7 @@ public Node getRoot() {
 
     @Override
     public <N extends Node> N createNode(final ItemModel itemModel) {
+        if (itemModel == null) return null;
```

Comment:
If `null` is a possible result of the `creaeNode` function, why not use `Optional`?

Something like this:
```
public Optional<<N extends Node> N> createNode(final ItemModel itemModel) {
        if (itemModel == null) return Optional.empty();
        final ProjectorNodeFactory factory = factories.get(itemModel.getClass());
        if (factory == null) {
            throw new IllegalArgumentException("No factory found for " + itemModel.getClass());
        }
        return Optional.ofNullable((N) factory.create(this, itemModel));
    }
```

Reply:
I changed the return value to optional and reverted that. I didn't like the resulting code - have a look at the invocations of createNode(). Using an optional changes the idea behind createNode(). The return value of createNode() is not allowed to be null, except the itemModel is null. This is a very clear relationship, that should not be weakened by wrapping the null parameter in an optional - IMHO